### PR TITLE
Added docs for trigger interface

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
I added a bunch of docs I felt were missing

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

File: docs/book/how-to/build-pipelines/compose-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+description: Compose your ZenML pipelines.
+---
+
+# Compose pipelines (reuse steps between pipelines)
```
Comment (strickvl) [inline docs/book/how-to/build-pipelines/compose-pipelines.md:5]:
```suggestion
description: Reuse steps between pipelines.
---

# Compose pipelines
```

File: docs/book/how-to/build-pipelines/compose-pipelines.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+@pipeline
+def training_pipeline():
+    training_data = data_loading_pipeline(mode="train")
+    model = training_step(data=training_data)
+    test_data = data_loading_pipeline(mode="test")
+    evaluation_step(model=model, data=test_data)
+```
+
+{% hint style="info" %}
```
Comment (strickvl) [inline docs/book/how-to/build-pipelines/compose-pipelines.md:32]:
```suggestion
```

Here we are calling one pipeline from within another pipeline, so functionally the `data_loading_pipeline` is functioning as a step within the `training_pipeline` though when it runs it is executing as an entirely separate pipeline. These two pipelines will be visible as separate pipelines within the ZenML dashboard, but practically they are working together within the umbrella of the `training_pipeline`.

{% hint style="info" %}
```

File: docs/book/how-to/build-pipelines/configuring-a-pipeline-at-runtime.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+description: Configuring a pipeline at runtime.
+---
+
+# Runtime configuration of a pipeline run
+
+It is often the case that there is a need to run a pipeline with a different configuration.
+In this case, you should in most cases use the `pipeline.with_options` method.  You can do
```
Comment (strickvl) [inline docs/book/how-to/build-pipelines/configuring-a-pipeline-at-runtime.md:8]:
```suggestion
In this case, you should in most cases use the [`pipeline.with_options()`](LINK GOES HERE) method.  You can do
```

Link to SDK docs

File: docs/book/how-to/build-pipelines/configuring-a-pipeline-at-runtime.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+this:
+
+1. Either by explicitly configuring options like `with_options(steps="trainer": {"parameters": {"param1": 1}})`
+2. Or by passing a YAML file using `with_options(config_file="path_to_yaml_file")`.
+
+You can learn more about these options [here](../use-configuration-files/README.md).
+
+However, there is one exception: If you would like to trigger a pipeline from the client
```
Comment (strickvl) [inline docs/book/how-to/build-pipelines/configuring-a-pipeline-at-runtime.md:16]:
```suggestion
However, there is one exception: if you would like to trigger a pipeline from the client
```

File: docs/book/how-to/build-pipelines/configuring-a-pipeline-at-runtime.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+or another pipeline, you would need to pass the PipelineRunConfiguration object.
```
Comment (strickvl) [inline docs/book/how-to/build-pipelines/configuring-a-pipeline-at-runtime.md:17]:
```suggestion
or another pipeline, you would need to pass the `PipelineRunConfiguration` object.
```

**Review by strickvl — CHANGES_REQUESTED** [review]

File: docs/book/how-to/build-pipelines/compose-pipelines.md (hunk 2 above)
Comment (htahir1) [reply docs/book/how-to/build-pipelines/compose-pipelines.md:32]: @strickvl not sure what you intended here, but I tried to fix it

Comment (strickvl) [reply docs/book/how-to/build-pipelines/compose-pipelines.md:32]: oh I think the extra backticks messed with the change. I just added some explanation. Let me take a look...

**Review by strickvl — APPROVED** [review]

File: docs/book/how-to/build-pipelines/compose-pipelines.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+Here we are calling one pipeline from within another pipeline, so functionally the `data_loading_pipeline` is functioning as a step within the `training_pipeline` though when it runs it is executing as an entirely separate pipeline. These two pipelines will be visible as separate pipelines within the ZenML dashboard, but practically they are working together within the umbrella of the `training_pipeline`. In order to trigger a pipeline from another, see [here](../trigger-pipelines/trigger-a-pipeline-from-another.md)
```
Comment (schustmi) [inline docs/book/how-to/build-pipelines/compose-pipelines.md:33]: `though when it runs it is executing as an entirely separate pipeline` It is not though right, it just adds the steps to `training_pipeline`?

Comment (schustmi) [reply docs/book/how-to/build-pipelines/compose-pipelines.md:33]: Actually this entire paragraph is wrong. They will not show up as separate pipelines

File: docs/book/how-to/trigger-pipelines/trigger-a-pipeline-from-rest-api.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+This is a [ZenML Pro](https://zenml.io/pro) only feature. Please [sign up here](https://cloud.zenml.io) get access.
+OSS users can only trigger a pipeline by calling the pipeline function inside their runner script.
+{% endhint %}
+
+Triggering a pipeline from the REST API **only** works with pipelines that are configured with a remote stack
+(i.e. at least a remote orchestrator, artifact store, and container registry)
+
+There are three calls that need to be made in order to trigger a pipeline from the REST API:
+
+1. `GET /pipelines?name=<PIPELINE_NAME>` -> This returns the <PIPELINE_ID>
```
Comment (schustmi) [inline docs/book/how-to/trigger-pipelines/trigger-a-pipeline-from-rest-api.md:18]: This returns a list of responses, which they need to parse the ID from

File: docs/book/how-to/trigger-pipelines/trigger-a-pipeline-from-rest-api.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+2. `GET /pipeline_builds?pipeline_id=<PIPELINE_ID>` -> This returns the build ID
```
Comment (schustmi) [inline docs/book/how-to/trigger-pipelines/trigger-a-pipeline-from-rest-api.md:19]: This returns a list again, which they need to parse the ID from

File: docs/book/how-to/trigger-pipelines/trigger-a-pipeline-from-rest-api.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+3. `POST /pipeline_builds/<BUILD_ID>/runs` -> This runs the pipeline. You can pass the [PipelineRunConfiguration](https://sdkdocs.zenml.io/latest/core_code_docs/core-config/#zenml.config.pipeline_run_configuration.PipelineRunConfiguration) in the body
+
+The above set of REST calls means that you can only trigger a pipeline that has been [built before](../customize-docker-builds/reuse-docker-builds.md).
```
Comment (schustmi) [inline docs/book/how-to/trigger-pipelines/trigger-a-pipeline-from-rest-api.md:22]: We should probably mention again that this needs a build with a remote stack etc.

File: docs/book/how-to/trigger-pipelines/trigger-a-pipeline-from-client.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
+---
+description: >-
+  Trigger a pipeline from the rest API.
```
Comment (schustmi) [inline docs/book/how-to/trigger-pipelines/trigger-a-pipeline-from-client.md:3]: client

File: docs/book/how-to/trigger-pipelines/trigger-a-pipeline-from-client.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+---
+
+# Trigger a pipeline from another
```
Comment (schustmi) [inline docs/book/how-to/trigger-pipelines/trigger-a-pipeline-from-client.md:6]: client

**Review by schustmi — CHANGES_REQUESTED** [review]

File: docs/book/how-to/build-pipelines/compose-pipelines.md (hunk 6 above)
Comment (strickvl) [reply docs/book/how-to/build-pipelines/compose-pipelines.md:33]: TIL :)

Comment (htahir1) [reply docs/book/how-to/build-pipelines/compose-pipelines.md:33]: I just used what @strickvl said :D

File: docs/book/how-to/build-pipelines/compose-pipelines.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+@pipeline
+def training_pipeline():
+    training_data = data_loading_pipeline(mode="train")
+    model = training_step(data=training_data)
+    test_data = data_loading_pipeline(mode="test")
+    evaluation_step(model=model, data=test_data)
+```
+
+{% hint style="info" %}
+Here we are calling one pipeline from within another pipeline, so functionally the `data_loading_pipeline` is functioning as a step within the `training_pipeline`, i.e. the steps of the former are added to the latter. The parent pipeline will be visible in the dashboard only. In order to actually trigger a pipeline from another, see [here](../trigger-pipelines/trigger-a-pipeline-from-another.md)
```
Comment (schustmi) [inline docs/book/how-to/build-pipelines/compose-pipelines.md:33]:
```suggestion
Here we are calling one pipeline from within another pipeline, so functionally the `data_loading_pipeline` is functioning as a step within the `training_pipeline`, i.e. the steps of the former are added to the latter. Only the parent pipeline will be visible in the dashboard. In order to actually trigger a pipeline from another, see [here](../trigger-pipelines/trigger-a-pipeline-from-another.md)
```

**Review by schustmi — APPROVED** [review]

**Review by strickvl — APPROVED** [review]

**Pull request merged**